src/main/java/com/github/manolo8/darkbot/gui/drawables/DynamicEntitiesDrawer.java

```diff
@@ -133,21 +135,44 @@ private void drawNpcs(MapGraphics mg) {
     private void drawPlayers(MapGraphics mg) {
         for (Player player : players) {
             boolean isEnemy = player.getEntityInfo().isEnemy();
-            mg.setColor(isEnemy ? "enemies" : "allies");
-            drawEntity(mg, player, 4, false);
 
+            int level = -1;
+            String color = isEnemy ? "enemies" : "allies";
```

Comment:
should probably cache the actual colors to avoid the N lookups when changing from one color to the other.

```
Color enemy = mg.getColor("enemy");
Color allies = mg.getColor("allies");
Color group = mg.getColor("group");

for (Player player : players) {
  // ...
  mg.setColor(isGroup ? group : isEnemy ? enemies : allies);
  // ...
}
```